src/library/scala/collection/mutable/EfficientMapBuilder.scala

```diff
@@ -0,0 +1,17 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+package scala.collection.mutable
+
+private[collection] trait EfficientMapBuilder[K, V] extends ((K, V, (K, V)) => Unit) {
```

Comment:
:-)
This is probably useful in several other places

e.g. for the caller to `override def ++=(xs: TraversableOnce[(A, B)]):` in HashMapBuilder, conversion from `Map4`, `MapBuilder` ( when it switches to use a `HashMapBuilder`), and for anything else that is a `HasForEachEntry`